datanode/api/trading_data_v2.go

```diff
@@ -3048,6 +3055,78 @@ func (t *tradingDataServiceV2) GetVegaTime(ctx context.Context, _ *v2.GetVegaTim
 
 // -- NetworkHistory --.
 
+func (t *tradingDataServiceV2) ExportNetworkHistory(req *v2.ExportNetworkHistoryRequest, stream v2.TradingDataService_ExportNetworkHistoryServer) error {
+	defer metrics.StartAPIRequestAndTimeGRPC("ExportNetworkHistory")()
+
+	if req.TableName == "" {
+		return formatE(ErrNetworkHistoryNoTableName, errors.New("empty table name"))
+	}
+
+	segments, err := t.networkHistoryService.ListAllHistorySegments()
+	if err != nil {
+		return formatE(ErrListAllNetworkHistorySegment, err)
+	}
+
+	ch, err := networkhistory.GetContiguousHistory(segments, req.FromBlock, req.ToBlock)
+	if err != nil {
+		return formatE(ErrNetworkHistoryGetContiguousSegments, err)
+	}
+
+	header := metadata.Pairs("Content-Disposition", fmt.Sprintf("attachment;filename=vega-%s-%06d-%06d.csv", req.TableName, ch.HeightFrom, ch.HeightTo))
+	if err := stream.SendHeader(header); err != nil {
+		return formatE(ErrSendingGRPCHeader, err)
+	}
+
+	var chunk bytes.Buffer
+	rowNum := 1
+
+	for _, segment := range ch.SegmentsOldestFirst {
+		outerTarFileReader, err := t.networkHistoryService.GetHistorySegmentReader(stream.Context(), segment.GetHistorySegmentId())
+		if err != nil {
+			return formatE(ErrNetworkHistoryOpeningSegment, err)
+		}
+
+		segmentData, err := fsutil.ReadNetworkHistorySegmentData(outerTarFileReader, req.TableName)
+		if err != nil {
+			return formatE(ErrNetworkHistoryExtractingSegment, err)
+		}
+
+		scanner := bufio.NewScanner(segmentData)
+		for ; scanner.Scan(); rowNum++ {
+			row := scanner.Text()
+			row = strings.Replace(row, "\t", ",", -1) // postgres dump file separated by tabs
+			row = strings.Replace(row, `\\x`, "", -1) // get rid of the \x prefix from BYTEA columns
+			chunk.WriteString(row)
+			chunk.WriteByte('\n')
```

Comment:
Also, on the csv column headers, probably easiest way would be to add this information into each segment when it is created, as a sort of meta file or something as will be relatively very small, not an issue.  And then this tool can optionally return with or without headers in the csv.

Reply:
ok, well if postgres handles that on import, thats the easiest for sure.   WRT schema version, this is already in the segment, I think just putting it in the file name would be sufficient (and then splitting data over schema boundary as per above comment)

Reply:
So I've semi-resolved this by allowing multiple schema versions in the same file, effectively concatenating two CSVs together. I'd rather not expose users to our schema versions and what have you, it's a bit of an implementation detail - here's the comment I wrote:

```
    // The first row of the CSV file is a header which allows you to identify the columns
    // of subsequent rows.
    //
    // It is possible that your (from_block, to_block) request spans over a range of blocks
    // in which the underlying schema of the tables changes. For example, a column may have
    // been added, removed, or renamed.
    //
    // If this happens, at the point of the schema migration the CSV file will contain a 
    // single blank line, and then a new set of headers, followed by the data in the new schema.
    //
    // An example of a CSV file containing such a schema change would look like:
    //
    // timestamp, value
    // 1, foo
    // 2, bar
    // 
    // timestamp, value, extra_value
    // 3, baz, apple
    //
```